lib/jekyll/configuration.rb

```diff
@@ -177,6 +177,7 @@ def read_config_files(files)
 
       begin
         files.each do |config_file|
+          next if config_file.nil? or config_file.empty?
```

Comment:
This is overly defensive while being non-defensive at the same time.  What I mean is it's better to just do `next unless config_file` instead of the empty because what happens if they send true? Or a symbol, or otherwise.  It might all still be toppling down.